### ISSUE TITLE
Integration with Janus

### ISSUE THREAD

**Issue by divergentdave** [divviup/divviup-api#84]
I'm working on integration tests between Janus and divviup-api in divviup/janus#1343, and I've just gotten them to pass on some hacky branches. Here's a list of issues I found.

- [x] In order to provision a task to both a helper and a leader using divviup-api, we need to be able to set the TaskId, VDAF verify key, and aggregator auth token. These are currently chosen randomly inside Janus's aggregator API. Additionally, we need to know the collector auth token, which is not currently returned in divviup-api's response. divviup/janus#1352 will add all four task parameters as optional fields in the aggregator API request to fix these issues, and assuming we take that change, divviup-api will need to pass those request values through. #107 
- [x] #88
- [x] #95
- [x] `max_batch_query_count` should be set to 1, not 0.
- [x] Wire formats for query type do not match. Janus expects Serde's default enum encoding, under the `query_type` key. divviup-api includes `max_batch_size` as an optional top-level field.
- [x] #90

**Comment by jbr** [comment]
I'm surprised by some of the encoding differences because the types are copy-pasted from the aggregator api implementation. I'll take a look at where that diverged

**Comment by divergentdave** [comment]
Some of the relevant structs (particularly HpkePublicKey) have handwritten `Serialize`/`Deserialize` implementations in other crates, instead of relying on derived implementations. These were originally written for the YAML format we use for manual task provisioning, and they're being used in a `serde_json` context here.

**Comment by jbr** [comment]
>  Wire formats for query type do not match. Janus expects Serde's default enum encoding, under the query_type key. divviup-api includes max_batch_size as an optional top-level field.
> The task entity code treats max_batch_size as an i32, while the database migration that created its column used a 64-bit integer.

Could you clarify these? There are several representations of tasks:
* [the database model](https://github.com/divviup/divviup-api/blob/main/src/entity/task.rs#L14-L33)
* [the NewTask accepted by POST ://divviup-api/api/tasks](https://github.com/divviup/divviup-api/blob/main/src/entity/task.rs#L65-L101)
* [the outbound TaskCreate sent to the aggregator api](https://github.com/divviup/divviup-api/blob/main/src/clients/aggregator_client/api_types.rs#L115-L126)
* [the inbound TaskResponse returned from the aggregator api](https://github.com/divviup/divviup-api/blob/main/src/clients/aggregator_client/api_types.rs#L128-L146)

Which of these has the wrong query type and max batch size?

**Comment by divergentdave** [comment]
Looking at the first query type issue again, the TaskCreate message is actually fine. We may want to change the NewTask message in order to better handle introduction of future query types, but that's not strictly required for integration test bringup.

The second query type issue is a mismatch between the database model and the database migration. I think this won't reproduce with SQLite because of its loose typing. When I ran a fixed size test, I got a Postgres error about INT8 not matching INT4.

Yes the aggregator_endpoints array needs to have exactly two URLs, with the leader coming first. (so the order will depend on the task's role parameter) Leader aggregators will use both URLs, one for outgoing HTTP requests to the helper and one for Location headers that point back to itself.
